design/service-indicator-lights.md

```diff
@@ -0,0 +1,201 @@
+<!--
+ This work is licensed under a Creative Commons Attribution 3.0
+ Unported License.
+
+ http://creativecommons.org/licenses/by/3.0/legalcode
+-->
+
+# service-indicator-lights
+
+## Status
+
+One of: provisional
+
+## Table of Contents
+
+<!--ts-->
+   * [service-indicator-lights](#service-indicator-lights)
+      * [Status](#status)
+      * [Table of Contents](#table-of-contents)
+      * [Summary](#summary)
+      * [Motivation](#motivation)
+         * [Goals](#goals)
+         * [Non-Goals](#non-goals)
+      * [Proposal](#proposal)
+         * [User Stories](#user-stories)
+            * [As an operator, I want to be able to match a software-reported fault to the hardware device that has failed.](#as-an-operator-i-want-to-be-able-to-match-a-software-reported-fault-to-the-hardware-device-that-has-failed)
+         * [Implementation Details/Notes/Constraints](#implementation-detailsnotesconstraints)
+         * [Risks and Mitigations](#risks-and-mitigations)
+      * [Design Details](#design-details)
+         * [Work Items](#work-items)
+         * [Dependencies](#dependencies)
+         * [Test Plan](#test-plan)
+         * [Upgrade / Downgrade Strategy](#upgrade--downgrade-strategy)
+         * [Version Skew Strategy](#version-skew-strategy)
+      * [Drawbacks](#drawbacks)
+      * [References](#references)
+
+<!-- Added by: dhellmann, at: Wed Jun 12 16:19:59 EDT 2019 -->
+
+<!--te-->
+
+## Summary
+
+Many hardware platforms have "indicator" lights in the chassis front
+(or rear) panel and sometimes have other lights associated with
+individual components like drives inside the chassis. We need to
+provide a way to turn those lights on when service is needed for a
```

Comment:
The light can be turned ON by hardware itself to indicate a failure. In that case we may want to turn it OFF for unit identification purposes.

Reply:
How would turning off the light help? Or do you just mean after the problem is resolved?

Reply:
No, I mean if we have many drives lighting up their LEDs at the same time to indicate failures. The operator wants to tell one failed drive from the other. The only (?) way to do that is to turn **off** the LEDs one by one to find out which drive is where.

Reply:
Ah, I see. Well, they will be able to do that by creating the resource and setting the flag to turn the light off.